src/Penrose/Util.hs

```diff
@@ -77,18 +77,27 @@ epsd = 10 ** (-10)
 -- General helper functions
 type Interval = (Float, Float) -- which causes type inference problems in Style for some reason.
 
+linspace :: (Autofloat a) => Int -> (a, a) -> [a]
```

Comment:
I think there is an off-by-one error with this function where it oversamples by one increment. I'd recommend using `lerpN` instead. (Note that the "API" for `lerpN` is slightly different; the `n` there is the number of interior points, and it auto-includes the endpoints.) For example:

```
λ> linspace 2 (0, 1)                                                                                                                                                                                                                                                          
[0.0,1.0,2.0]                                                                                                                                                                                                                                                                 
λ> linspace 3 (0, 1)                                                                                                                                                                                                                                                          
[0.0,0.5,1.0,1.5]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
λ> linspace 4 (0, 1)                                                                                                                                                                                                                                                          
[0.0,0.3333333333333333,0.6666666666666666,1.0,1.3333333333333333]                                                                                                                                                                                                            
λ> linspace 3 (-2, 2)                                                                                                                                                                                                                                                         
[-2.0,0.0,2.0,4.0]                                                                                                                                                                                                                                                            
λ> lerpN 0 1 2                                                                                                                                                                                                                                                                
[0.0,0.3333333333333333,0.6666666666666666,1.0]                                                                                                                                                                                                                               
λ> lerpN 0 1 1                                                                                                                                                                                                                                                                
[0.0,0.5,1.0]              
```

BTW, I think you already know how to load functions but just in case:

```
stack ghci [--profile]
:l Penrose.Util
```

Reply:
Good catch. I'll use your function then.